testing/functional_tests/test_automate_training.py

```diff
@@ -34,6 +35,38 @@ def test_automate_training(download_functional_test_files, script_runner):
     check_sha256(str(file_config))
 
 
+def test_automate_training_run_test_debug(download_functional_test_files):
+    """A unit test similar to test_automate_training_run_test but allow step through (instead of using script caller/
+    subprocess mode which cannot be stepped. Other than that, nothing else really changed and is exactly the same.
+    Very useful for debugging this high level function to spot problems
+
+    Fixture Required:
+        download_functional_test_files:
+    """
+    file_config = os.path.join(__data_testing_dir__, 'automate_training_config.json')
+    file_config_hyper = os.path.join(__data_testing_dir__,
+                                     'automate_training_hyperparameter_opt.json')
+    __output_dir__ = os.path.join(__tmp_dir__, 'results')
```

Comment:
Here we use "os" whereas in `test_automate_training_run_test` below, we use "Path". Is there a preferred method? I would suggest using the same if possible.

Reply:
Agreed, I suggest we go with `Path` for maintaining consistency. See also [this](https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/) link in favor of using `pathlib`, in general.

Reply:
Good point! Updated.